pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -49,15 +52,20 @@ func (d *DualWriterMode1) Create(ctx context.Context, original runtime.Object, c
 	}
 	d.recordLegacyDuration(false, mode1Str, options.Kind, method, startLegacy)
 
+	d.pendingActions.Add(1)
 	go func() {
+		defer d.pendingActions.Done()
+		ctx := context.WithoutCancel(ctx)
 		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage create timeout"))
+		defer cancel()
+		created := created.DeepCopyObject()
```

Comment:
If we don't make a deep copy, we will be modifying the returned object instead when we call `enrichLegacyObject` below. And this would be done in a different goroutine, which means that what is sent to the client is undefined, since we don't know if it will be the response from legacy, the version enriched or something in the middle since it may be in the process of being written.